src/bin/cql-stress-cassandra-stress/operation/mixed.rs

```diff
@@ -0,0 +1,169 @@
+        let write_operation_factory =
+            WriteOperationFactory::new(settings.clone(), session.clone()).await?;
+        let counter_write_operation_factory =
+            CounterWriteOperationFactory::new(settings.clone(), session.clone()).await?;
+        let read_operation_factory =
+            RegularReadOperationFactory::new(settings.clone(), session.clone(), DEFAULT_TABLE_NAME)
+                .await?;
+        let counter_read_operation_factory =
+            CounterReadOperationFactory::new(settings.clone(), session, DEFAULT_COUNTER_TABLE_NAME)
+                .await?;
+
+        Ok(Self {
+            settings,
+            write_operation_factory,
+            counter_write_operation_factory,
+            read_operation_factory,
+            counter_read_operation_factory,
+            operation_ratio,
+            workload_factory,
+            max_operations,
+            stats,
+        })
+    }
+}
+
+make_runnable!(MixedOperation);
+impl MixedOperation {
+    async fn execute(&mut self, ctx: &OperationContext) -> Result<ControlFlow<()>> {
+        if self
+            .max_operations
+            .is_some_and(|max_ops| ctx.operation_id >= max_ops)
+        {
+            return Ok(ControlFlow::Break(()));
+        }
+
+        if self.current_operation_count == 0 {
+            self.current_operation = self.operation_ratio.sample();
+            self.current_operation_count =
+                (self.clustering_distribution.next_i64() as usize).max(1);
+        }
+
+        let result = match &self.current_operation {
+            MixedSubcommand::Read => {
+                let row = self
+                    .cached_row
+                    .get_or_insert_with(|| self.read_operation.generate_row(&mut self.workload));
+                self.read_operation.execute(row).await
+            }
+            MixedSubcommand::CounterRead => {
+                let row = self.cached_row.get_or_insert_with(|| {
+                    self.counter_read_operation.generate_row(&mut self.workload)
+                });
+                self.counter_read_operation.execute(row).await
+            }
+            MixedSubcommand::Write => {
+                let row = self
+                    .cached_row
+                    .get_or_insert_with(|| self.write_operation.generate_row(&mut self.workload));
+                self.write_operation.execute(row).await
+            }
+            MixedSubcommand::CounterWrite => {
+                let row = self.cached_row.get_or_insert_with(|| {
+                    self.counter_write_operation
+                        .generate_row(&mut self.workload)
+                });
+                self.counter_write_operation.execute(row).await
+            }
+        };
+
+        self.stats.get_shard_mut().account_operation(ctx, &result);
+
+        if result.is_ok() {
+            self.current_operation_count = self.current_operation_count.saturating_sub(1);
```

Comment:
Is there a particular reason here for using saturating_sub? We check for `self.current_operation_count == 0` earlier in the function, so the current operation count here should not be `0`, right?

Reply:
Yes, it's pointless to have saturating_sub here. It remains from the previous version where we would have:
```rust
self.current_operation_count = self.clustering_distribution.next_i64() as usize;
```
instead of
```rust
self.current_operation_count =
    (self.clustering_distribution.next_i64() as usize).max(1);
```

In the previous version there was a possibility that `self.current_operation_count` was 0 after taking the sample. Now we are guaranteed that it's at least 1 (thanks to `usize::max` method).

I'll get rid of the `saturating_sub`.